reverse_only_letters.py

```diff
@@ -0,0 +1,44 @@
+#Problem: https://leetcode.com/problems/reverse-only-letters/
+
+# Time complexity: O(N)
+# Memory complexity: O(N)
+#Method 1:
+def reverse_letters(s):
+    chars = []
+    reverse_letters = []
+    for ch in s:
+        if ch.isalpha():
+            chars.append(ch)
+
+    for ch in s:
+        if ch.isalpha():
+            reverse_letters.append(chars.pop())
+        else:
+            reverse_letters.append(ch)
+
+    return ''.join(reverse_letters)
+            
+
+# Time complexity: O(N)
+# Memory complexity: O(N)
```

Comment:
I want you to think how you could reduce this memory complexity as it should be possible to go lower.